client/client.go

```diff
@@ -397,8 +393,8 @@ func (c *Client) BalanceProofWatcher() {
 
 // FrozenWatcher listens for Frozen events and calls WithdrawFrozen
 // if a balance proof is available.
-func (c *Client) FrozenWatcher() {
-	ctx, cancel := context.WithCancel(c.Ctx())
+func (c *Client) FrozenWatcher(_ctx context.Context) {
+	ctx, cancel := context.WithCancel(_ctx)
 	defer cancel()
```

Comment:
you already pass a cancel-context whose cancel function you set as the watcher canceller. So you don't need this cancel wrapper.

But I think it's actually better to not pass a context to the `FrozenWatcher` at all and set
`ctx, c.stopFrozenWatcher = context.WithCancel(c.Ctx())` here directly.

Reply:
Yes good idea